[//]: # "File generated from a template. Do not edit this file directly."

# node-param-default-wrong-for-boolean

`default` for boolean-type node parameter must be a boolean.

📋 This rule is part of the `plugin:flowease-nodes-base/nodes` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "boolean",
	default: "",
};
```

✅ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "boolean",
	default: false,
};
```

## Links

- [Rule source](../../lib/rules/node-param-default-wrong-for-boolean.ts)
- [Test source](../../tests/node-param-default-wrong-for-boolean.test.ts)
