[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-excess-inner-whitespace

`displayName` in node parameter or in fixed collection section must not contain excess inner whitespace. Applicable by extension to `name` in options-type or multi-options-type node parameter.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test   Test",
	name: "test",
	type: "string",
	default: "",
	description: "This a sentence",
};

const test = {
	name: "Test    Test",
	value: "test",
	description: "This is a sentence",
};

const test = {
	displayName: "Test",
	name: "test",
	type: "fixedCollection",
	default: "a",
	options: [
		{
			displayName: "Details   Test",
			name: "detailsTest",
			values: [
				{
					displayName: "A",
					name: "a",
					type: "string",
					default: "",
				},
			],
		},
	],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test Test",
	name: "test",
	type: "string",
	default: "",
	description: "This a sentence",
};

const test = {
	name: "Test Test",
	value: "test",
	description: "This is a sentence",
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-excess-inner-whitespace.ts)
- [Test source](../../tests/node-param-display-name-excess-inner-whitespace.test.ts)
