[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-missing-from-limit

`description` in Limit node parameter must be present.

📋 This rule is part of the `plugin:flowease-nodes-base/nodes` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
	displayName: "Limit",
	name: "limit",
	default: 50,
	typeOptions: {
		minValue: 1,
	},
	type: "number",
};
```

✅ Example of **correct** code:

```js
const test = {
	displayName: "Limit",
	name: "limit",
	default: 50,
	typeOptions: {
		minValue: 1,
	},
	type: "number",
	description: "Max number of results to return",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-missing-from-limit.ts)
- [Test source](../../tests/node-param-description-missing-from-limit.test.ts)
