[//]: # "File generated from a template. Do not edit this file directly."

# node-param-required-false

`required: false` in node parameter must be removed because it is implied.

📋 This rule is part of the `plugin:flowease-nodes-base/nodes` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	required: false,
};
```

✅ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	required: true,
};
```

## Links

- [Rule source](../../lib/rules/node-param-required-false.ts)
- [Test source](../../tests/node-param-required-false.test.ts)
