[//]: # "File generated from a template. Do not edit this file directly."

# node-param-default-wrong-for-limit

`default` for a Limit node parameter must be `50`.

📋 This rule is part of the `plugin:flowease-nodes-base/nodes` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
	displayName: "Limit",
	name: "limit",
	type: "number",
	default: 10,
};
```

✅ Example of **correct** code:

```js
const test = {
	displayName: "Limit",
	name: "limit",
	type: "number",
	default: 50,
};
```

## Links

- [Rule source](../../lib/rules/node-param-default-wrong-for-limit.ts)
- [Test source](../../tests/node-param-default-wrong-for-limit.test.ts)
