[//]: # "File generated from a template. Do not edit this file directly."

# node-param-color-type-unused

`string`-type color-related node parameter must be `color`-type.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Foreground Colour",
	name: "foregroundColor",
	type: "string",
	default: "#000000",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Foreground Colour",
	name: "foregroundColor",
	type: "color",
	default: "#000000",
};

const test = {
	displayName: "Color",
	name: "color",
	type: "options",
	default: "blue",
	options: [
		{
			name: "Blue",
			value: "blue",
		},
		{
			name: "Green",
			value: "green",
		},
	],
};
```

## Links

- [Rule source](../../lib/rules/node-param-color-type-unused.ts)
- [Test source](../../tests/node-param-color-type-unused.test.ts)
