[//]: # "File generated from a template. Do not edit this file directly."

# node-param-collection-type-item-required

Items in collection-type node parameter must not have a `required` property.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Options",
	name: "options",
	type: "collection",
	placeholder: "Add Option",
	default: {},
	displayOptions: {
		show: {
			resource: ["collection"],
			operation: ["getAll"],
		},
	},
	options: [
		{
			displayName: "First",
			name: "first",
			type: "boolean",
			required: true,
			default: true,
		},
		{
			displayName: "Second",
			name: "second",
			type: "string",
			default: "",
		},
	],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Options",
	name: "options",
	type: "collection",
	placeholder: "Add Option",
	default: {},
	displayOptions: {
		show: {
			resource: ["collection"],
			operation: ["getAll"],
		},
	},
	options: [
		{
			displayName: "First",
			name: "first",
			type: "boolean",
			default: true,
		},
		{
			displayName: "Second",
			name: "second",
			type: "string",
			default: "",
		},
	],
};
```

## Links

- [Rule source](../../lib/rules/node-param-collection-type-item-required.ts)
- [Test source](../../tests/node-param-collection-type-item-required.test.ts)
