[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-miscased-id

`ID` in `description` in node parameter must be fully uppercased. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "The Id of the user provided",
};

const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "The Ids of the user",
};

const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "The id of the user",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The id of the user",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The Id of the user",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "The ID of the user",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The ID of the user",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "Unique identifier",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-miscased-id.ts)
- [Test source](../../tests/node-param-description-miscased-id.test.ts)
