[//]: # "File generated from a template. Do not edit this file directly."

# node-param-option-description-identical-to-name

`description` in option in options-type node parameter must not be identical to `name`.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	name: "Full Name",
	value: "fullName",
	description: "The full name",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	name: "Full Name",
	value: "fullName",
	description: "First and last name of the user",
};
```

## Links

- [Rule source](../../lib/rules/node-param-option-description-identical-to-name.ts)
- [Test source](../../tests/node-param-option-description-identical-to-name.test.ts)
