[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-wrong-for-dynamic-options

`displayName` for dynamic-options-type node parameter must end with `Name or ID`

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Field ID",
	name: "field",
	type: "options",
	typeOptions: {
		loadOptionsMethod: "getFields",
	},
	default: "",
};

const test = {
	displayName: "Field Name",
	name: "field",
	type: "options",
	typeOptions: {
		loadOptionsMethod: "getFields",
	},
	default: "",
};

const test = {
	displayName: "Field Name or",
	name: "field",
	type: "options",
	typeOptions: {
		loadOptionsMethod: "getFields",
	},
	default: "",
};

const test = {
	displayName: "Field Name/ID",
	name: "field",
	type: "options",
	typeOptions: {
		loadOptionsMethod: "getFields",
	},
	default: "",
};

const test = {
	displayName: "Field ID/Name",
	name: "field",
	type: "options",
	typeOptions: {
		loadOptionsMethod: "getFields",
	},
	default: "",
};

const test = {
	displayName: "Field Names/IDs",
	name: "field",
	type: "options",
	typeOptions: {
		loadOptionsMethod: "getFields",
	},
	default: "",
};

const test = {
	displayName: "Field",
	name: "field",
	type: "options",
	typeOptions: {
		loadOptionsMethod: "getFields",
	},
	default: "",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Field Name or ID",
	name: "field",
	type: "options",
	typeOptions: {
		loadOptionsMethod: "getFields",
	},
	default: "",
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-wrong-for-dynamic-options.ts)
- [Test source](../../tests/node-param-display-name-wrong-for-dynamic-options.test.ts)
