[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-lowercase-first-char

First char in `description` in node parameter must be uppercase. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "this is a test",
};

const test = {
	name: "Username",
	value: "username",
	description: "the name of the user",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a test",
};

const test = {
	name: "Username",
	value: "username",
	description: "The name of the user",
};

const test = {
	displayName: "Format",
	name: "format",
	type: "options",
	options: [
		{
			name: "bmp",
			value: "bmp",
		},
		{
			name: "gif",
			value: "gif",
		},
		{
			name: "jpeg",
			value: "jpeg",
		},
		{
			name: "png",
			value: "png",
		},
		{
			name: "tiff",
			value: "tiff",
		},
	],
	default: "jpeg",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-lowercase-first-char.ts)
- [Test source](../../tests/node-param-description-lowercase-first-char.test.ts)
