[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-wrong-for-return-all

`description` for Return All node parameter must be `Whether to return all results or only up to a given limit`

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Return All",
	name: "returnAll",
	type: "boolean",
	default: true,
	description: "Wrong",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Return All",
	name: "returnAll",
	type: "boolean",
	default: true,
	description: "Whether to return all results or only up to a given limit",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-wrong-for-return-all.ts)
- [Test source](../../tests/node-param-description-wrong-for-return-all.test.ts)
