[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-excess-final-period

`description` in node parameter must end without a final period if a single-sentence description. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This a sentence.",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The ID of the user.",
};

const test = {
	displayName: "Address",
	name: "address",
	type: "string",
	default: "",
	description: "Person's email address.",
};

const test = {
	displayName: "Name",
	name: "name",
	type: "string",
	default: "",
	description: `Custom Field\'s name.`,
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This a sentence",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The ID of the user",
};

const test = {
	name: "Username",
	value: "username",
	description: "Start e.g. end",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-excess-final-period.ts)
- [Test source](../../tests/node-param-description-excess-final-period.test.ts)
