[//]: # "File generated from a template. Do not edit this file directly."

# node-param-operation-option-description-wrong-for-get-many

The property `description` in a Get Many option in an Operation node parameter must mention `many` instead of `all`.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "options",
	noDataExpression: true,
	displayOptions: {
		show: {
			resource: ["entity"],
		},
	},
	default: "getAll",
	options: [
		{
			name: "Get Many",
			value: "getAll",
			description: "Retrieve all entities",
			action: "Get many entities",
		},
	],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "options",
	noDataExpression: true,
	displayOptions: {
		show: {
			resource: ["entity"],
		},
	},
	default: "getAll",
	options: [
		{
			name: "Get Many",
			value: "getAll",
			description: "Retrieve many entities",
			action: "Get many entities",
		},
	],
};
```

## Links

- [Rule source](../../lib/rules/node-param-operation-option-description-wrong-for-get-many.ts)
- [Test source](../../tests/node-param-operation-option-description-wrong-for-get-many.test.ts)
