[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-unneeded-backticks

`description` in node parameter must not use unneeded backticks. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: `This a sentence`,
};

const test = {
	name: "User ID",
	value: "userId",
	description: `The ID of the user`,
};

const test = {
	name: "Timezone",
	value: "timezone",
	description: `Seatable server's timezone`,
};

const test = {
	name: "Timezone",
	value: "timezone",
	description: `AAA 'bbb' CCC`,
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This a sentence",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The ID of the user",
};

const test = {
	displayName: "Access Token",
	name: "accessToken",
	type: "string",
	default: "",
	description: `The access token must have the following scopes:
    <ul>
        <li>Create/modify webhooks</li>
        <li>View webhooks</li>
        <li>View surveys</li>
        <li>View collectors</li>
        <li>View responses</li>
        <li>View response details</li>
    </ul>`,
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-unneeded-backticks.ts)
- [Test source](../../tests/node-param-description-unneeded-backticks.test.ts)
