[//]: # "File generated from a template. Do not edit this file directly."

# node-param-default-wrong-for-string

`default` for a string-type node parameter must be a string, unless `typeOptions.multipleValues` is set to `true`.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: 1,
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
};

const test = {
	displayName: "Tags",
	name: "tags",
	type: "string",
	typeOptions: {
		multipleValues: true,
		multipleValueButtonText: "Add Tag",
	},
	default: [],
};

const test = {
	displayName: "Scope",
	name: "scope",
	type: "hidden",
	default: scopes,
};
```

## Links

- [Rule source](../../lib/rules/node-param-default-wrong-for-string.ts)
- [Test source](../../tests/node-param-default-wrong-for-string.test.ts)
