[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-boolean-without-whether

`description` in a boolean node parameter must start with `Whether`.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "boolean",
	default: true,
	description: "This is wrong",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "boolean",
	default: true,
	description: "Whether to do something",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-boolean-without-whether.ts)
- [Test source](../../tests/node-param-description-boolean-without-whether.test.ts)
