[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-url-missing-protocol

`description` in node parameter must include protocol e.g. `https://` when containing a URL. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: 'This is a <a href="github.com">link</a>',
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a <a href='github.com'>link</a>",
};

const test = {
	name: "Test",
	value: "test",
	description: 'This is a <a href="github.com">link</a>',
};

const test = {
	name: "Test",
	value: "test",
	description: "This is a <a href='github.com'>link</a>",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: 'This is a <a href="https://github.com">link</a>',
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a <a href='https://github.com'>link</a>",
};

const test = {
	name: "Test",
	value: "test",
	description: 'This is a <a href="https://github.com">link</a>',
};

const test = {
	name: "Test",
	value: "test",
	description: "This is a <a href='https://github.com'>link</a>",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-url-missing-protocol.ts)
- [Test source](../../tests/node-param-description-url-missing-protocol.test.ts)
