[//]: # "File generated from a template. Do not edit this file directly."

# node-param-option-name-wrong-for-get-many

Option `name` for Get Many node parameter must be `Get Many`

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	name: "List",
	value: "getAll",
};

const test = {
	name: "Get All",
	value: "getAll",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	name: "Get Many",
	value: "getAll",
};
```

## Links

- [Rule source](../../lib/rules/node-param-option-name-wrong-for-get-many.ts)
- [Test source](../../tests/node-param-option-name-wrong-for-get-many.test.ts)
