[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-untrimmed

`description` in node parameter must be trimmed. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "This is a description   ",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The ID of the user   ",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "This is a description",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The ID of the user",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-untrimmed.ts)
- [Test source](../../tests/node-param-description-untrimmed.test.ts)
