[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-missing-for-ignore-ssl-issues

`description` for Ignore SSL node parameter must be present.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Ignore SSL Issues",
	name: "allowUnauthorizedCerts",
	type: "boolean",
	default: true,
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Ignore SSL Issues",
	name: "allowUnauthorizedCerts",
	type: "boolean",
	description:
		"Whether to connect even if SSL certificate validation is not possible",
	default: true,
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-missing-for-ignore-ssl-issues.ts)
- [Test source](../../tests/node-param-description-missing-for-ignore-ssl-issues.test.ts)
