[//]: # "File generated from a template. Do not edit this file directly."

# node-param-default-wrong-for-number

`default` for a number-type node parameter must be a number, except for a number-type ID parameter.

📋 This rule is part of the `plugin:flowease-nodes-base/nodes` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "number",
	default: "0",
};
```

✅ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "number",
	default: 0,
};

const test = {
	displayName: "Test",
	name: "test",
	type: "number",
	default: "",
};

const test = {
	displayName: "Account Contact ID",
	name: "accountContactId",
	type: "number",
	default: "",
	required: true,
};
```

## Links

- [Rule source](../../lib/rules/node-param-default-wrong-for-number.ts)
- [Test source](../../tests/node-param-default-wrong-for-number.test.ts)
