[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-not-first-position

By convention, `displayName` in node parameter must be placed first.

📋 This rule is part of the `plugin:flowease-nodes-base/nodes` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
	name: "test",
	displayName: "Test",
	type: "string",
	default: "",
};
```

✅ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
};

const test = {
	name: "Test of Tests",
	value: "testOfTests",
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-not-first-position.ts)
- [Test source](../../tests/node-param-display-name-not-first-position.test.ts)
