[//]: # "File generated from a template. Do not edit this file directly."

# node-param-min-value-wrong-for-limit

`minValue` for Limit node parameter must be a positive integer.

📋 This rule is part of the `plugin:flowease-nodes-base/nodes` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
	displayName: "Limit",
	name: "limit",
	type: "number",
	default: 50,
	typeOptions: {
		minValue: 0,
	},
};
```

✅ Example of **correct** code:

```js
const test = {
	displayName: "Limit",
	name: "limit",
	type: "number",
	default: 50,
	typeOptions: {
		minValue: 1,
	},
};
```

## Links

- [Rule source](../../lib/rules/node-param-min-value-wrong-for-limit.ts)
- [Test source](../../tests/node-param-min-value-wrong-for-limit.test.ts)
