[//]: # "File generated from a template. Do not edit this file directly."

# node-class-description-credentials-name-unsuffixed

`name` under `credentials` in node class description must be suffixed with `-Api`.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
	description = {
		displayName: "Test",
		name: "test",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "This is a sentence",
		defaults: {
			name: "Test",
		},
		inputs: ["main"],
		outputs: ["main"],
		credentials: [
			{
				name: "testOAuth2",
			},
			{
				name: "testApi",
			},
		],
	};
}

class TestNode {
	description = {
		displayName: "Test",
		name: "test",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "This is a sentence",
		defaults: {
			name: "Test",
		},
		inputs: ["main"],
		outputs: ["main"],
		credentials: [
			{
				name: "testOAuth2Ap",
			},
			{
				name: "testApi",
			},
		],
	};
}

class TestNode {
	description = {
		displayName: "Test",
		name: "test",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "This is a sentence",
		defaults: {
			name: "Test",
		},
		inputs: ["main"],
		outputs: ["main"],
		credentials: [
			{
				name: "testOAuth2A",
			},
			{
				name: "testApi",
			},
		],
	};
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
	description = {
		displayName: "Test",
		name: "test",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "This is a sentence",
		defaults: {
			name: "Test",
		},
		inputs: ["main"],
		outputs: ["main"],
		credentials: [
			{
				name: "testOAuth2Api",
			},
			{
				name: "testApi",
			},
		],
	};
}
```

## Links

- [Rule source](../../lib/rules/node-class-description-credentials-name-unsuffixed.ts)
- [Test source](../../tests/node-class-description-credentials-name-unsuffixed.test.ts)
