[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-missing-for-return-all

`description` for Return All node parameter must be present.

📋 This rule is part of the `plugin:flowease-nodes-base/nodes` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
	displayName: "Return All",
	name: "returnAll",
	type: "boolean",
	default: true,
};
```

✅ Example of **correct** code:

```js
const test = {
	displayName: "Return All",
	name: "returnAll",
	type: "boolean",
	description: "Whether to return all results or only up to a given limit",
	default: true,
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-missing-for-return-all.ts)
- [Test source](../../tests/node-param-description-missing-for-return-all.test.ts)
