[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-missing-for-simplify

`description` for Simplify node parameter must be present.

📋 This rule is part of the `plugin:flowease-nodes-base/nodes` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
	displayName: "Simplify",
	name: "simple",
	type: "boolean",
	default: true,
};
```

✅ Example of **correct** code:

```js
const test = {
	displayName: "Simplify",
	name: "simple",
	type: "boolean",
	description:
		"Whether to return a simplified version of the response instead of the raw data",
	default: true,
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-missing-for-simplify.ts)
- [Test source](../../tests/node-param-description-missing-for-simplify.test.ts)
