[//]: # "File generated from a template. Do not edit this file directly."

# node-param-resource-without-no-data-expression

`noDataExpression` in a Resource node parameter must be present and enabled.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Resource",
	name: "resource",
	type: "options",
	noDataExpression: false,
	options: [
		{
			name: "User",
			value: "user",
		},
	],
	default: "user",
};

const test = {
	displayName: "Resource",
	name: "resource",
	type: "options",
	options: [
		{
			name: "User",
			value: "user",
		},
	],
	default: "user",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Resource",
	name: "resource",
	type: "options",
	noDataExpression: true,
	options: [
		{
			name: "User",
			value: "user",
		},
	],
	default: "user",
};
```

## Links

- [Rule source](../../lib/rules/node-param-resource-without-no-data-expression.ts)
- [Test source](../../tests/node-param-resource-without-no-data-expression.test.ts)
