[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-miscased-json

`JSON` in `description` in node parameter must be fully uppercased. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "JSON",
	name: "json",
	type: "string",
	default: "",
	description: "The json to send",
};

const test = {
	displayName: "JSON",
	name: "json",
	type: "string",
	default: "",
	description: "The Json to send",
};

const test = {
	name: "JSON",
	value: "json",
	description: "The json to send",
};

const test = {
	name: "JSON",
	value: "json",
	description: "The json to send. The json to send.",
};

const test = {
	name: "JSON",
	value: "json",
	description: "The Json to send",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "JSON",
	name: "json",
	type: "string",
	default: "",
	description: "The JSON to send",
};

const test = {
	name: "JSON",
	value: "json",
	description: "The JSON",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-miscased-json.ts)
- [Test source](../../tests/node-param-description-miscased-json.test.ts)
