[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-line-break-html-tag

`description` in node parameter must not contain an HTML line break. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a <br /> sentence",
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a <br > sentence",
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a <br> sentence",
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a </br> sentence",
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a </ br> sentence",
};

const test = {
	name: "Test",
	value: "test",
	description: "This is a <br /> sentence",
};

const test = {
	name: "Test",
	value: "test",
	description: "This is a <br/> sentence",
};

const test = {
	name: "Test",
	value: "test",
	description: "This is a <br> sentence",
};

const test = {
	name: "Test",
	value: "test",
	description: "This is a </br> sentence",
};

const test = {
	name: "Test",
	value: "test",
	description: "This is a </br > sentence",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a sentence",
};

const test = {
	displayName: "Private Key",
	name: "privateKey",
	type: "string",
	typeOptions: {
		password: true,
	},
	default: "",
	required: true,
	description:
		"Use the multiline editor. Make sure it is in standard PEM key format:<br />-----BEGIN PRIVATE KEY-----<br />KEY DATA GOES HERE<br />-----END PRIVATE KEY-----",
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a sentence.<br><br> This is another.",
};

const test = {
	name: "Test",
	value: "test",
	description: "This is a sentence",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-line-break-html-tag.ts)
- [Test source](../../tests/node-param-description-line-break-html-tag.test.ts)
