[//]: # "File generated from a template. Do not edit this file directly."

# node-param-option-value-duplicate

Option `value` in options-type node parameter must not be a duplicate.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "options",
	options: [
		{
			name: "A",
			value: "a",
		},
		{
			name: "B",
			value: "a",
		},
	],
	default: "a",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "options",
	options: [
		{
			name: "A",
			value: "a",
		},
		{
			name: "B",
			value: "b",
		},
	],
	default: "a",
};
```

## Links

- [Rule source](../../lib/rules/node-param-option-value-duplicate.ts)
- [Test source](../../tests/node-param-option-value-duplicate.test.ts)
