[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-unencoded-angle-brackets

`description` in node parameter must encode angle brackets for them to render. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a <value>",
};

const test = {
	name: "Test",
	value: "test",
	description: "This is a <value>",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "This is a <b>sentence</b>",
};

const test = {
	displayName: "Private Key",
	name: "privateKey",
	type: "string",
	typeOptions: {
		password: true,
	},
	default: "",
	required: true,
	description:
		"Use the multiline editor. Make sure it is in standard PEM key format:<br />-----BEGIN PRIVATE KEY-----<br />KEY DATA GOES HERE<br />-----END PRIVATE KEY-----",
};

const test = {
	name: "Test",
	value: "test",
	description: "This is a <b>sentence</b>",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-unencoded-angle-brackets.ts)
- [Test source](../../tests/node-param-description-unencoded-angle-brackets.test.ts)
