[//]: # "File generated from a template. Do not edit this file directly."

# node-class-description-empty-string

`description` in node class description must be filled out.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
	description = {
		displayName: "TestTrigger",
		name: "testTrigger",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "",
		defaults: {
			name: "Test",
		},
		inputs: ["main"],
		outputs: ["main"],
	};
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
	description = {
		displayName: "TestTrigger",
		name: "testTrigger",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "This is a sentence",
		defaults: {
			name: "Test",
		},
		inputs: ["main"],
		outputs: ["main"],
	};
}
```

## Links

- [Rule source](../../lib/rules/node-class-description-empty-string.ts)
- [Test source](../../tests/node-class-description-empty-string.test.ts)
