[//]: # "File generated from a template. Do not edit this file directly."

# node-class-description-display-name-unsuffixed-trigger-node

`displayName` in node class description for trigger node must be suffixed with `-Trigger`.

üìã This rule is part of the `plugin:flowease-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
	description = {
		displayName: "Test",
		name: "testTrigger",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "This is a sentence",
		defaults: {
			name: "Test",
		},
		inputs: ["main"],
		outputs: ["main"],
	};
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
	description = {
		displayName: "Test Trigger",
		name: "testTrigger",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "This is a sentence",
		defaults: {
			name: "Test",
		},
		inputs: ["main"],
		outputs: ["main"],
	};
}

class TestNode {
	description = {
		displayName: "Test Trigger (Beta)",
		name: "testTrigger",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "This is a sentence",
		defaults: {
			name: "Test",
		},
		inputs: ["main"],
		outputs: ["main"],
	};
}
```

## Links

- [Rule source](../../lib/rules/node-class-description-display-name-unsuffixed-trigger-node.ts)
- [Test source](../../tests/node-class-description-display-name-unsuffixed-trigger-node.test.ts)
